openupgrade_scripts/scripts/payment/16.0.2.0/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,213 @@
+---Models in module 'payment'---
+obsolete model payment.acquirer
+# TODO: Renamed to payment.provider
+
+obsolete model payment.acquirer.onboarding.wizard [transient]
+# NOTHING TO DO: Renamed to payment.provider.onboarding.wizard
+
+model payment.refund.wizard (moved to account_payment) [transient]
+# NOTHING TO DO: it's transient, it will be deleted from this module and added to the other one.
+
+new model payment.provider
+new model payment.provider.onboarding.wizard [transient]
+# See comment above.
+
+---Fields in module 'payment'---
+payment      / account.bank.statement.line / amount_paid (float)           : module is now 'account_payment' ('payment')
+payment      / account.bank.statement.line / authorized_transaction_ids (many2many): module is now 'account_payment' ('payment')
+payment      / account.bank.statement.line / transaction_ids (many2many)   : module is now 'account_payment' ('payment')
+payment      / account.move             / amount_paid (float)           : module is now 'account_payment' ('payment')
+payment      / account.move             / authorized_transaction_ids (many2many): module is now 'account_payment' ('payment')
+payment      / account.move             / transaction_ids (many2many)   : module is now 'account_payment' ('payment')
+payment      / account.payment          / amount_available_for_refund (float): module is now 'account_payment' ('payment')
+payment      / account.payment          / amount_paid (float)           : module is now 'account_payment' ('payment')
+payment      / account.payment          / authorized_transaction_ids (many2many): module is now 'account_payment' ('payment')
+payment      / account.payment          / payment_token_id (many2one)   : module is now 'account_payment' ('payment')
+payment      / account.payment          / payment_transaction_id (many2one): module is now 'account_payment' ('payment')
+payment      / account.payment          / refunds_count (integer)       : module is now 'account_payment' ('payment')
+payment      / account.payment          / source_payment_id (many2one)  : module is now 'account_payment' ('payment')
+payment      / account.payment          / suitable_payment_token_ids (many2many): module is now 'account_payment' ('payment')
+payment      / account.payment          / transaction_ids (many2many)   : module is now 'account_payment' ('payment')
+payment      / account.payment          / use_electronic_payment_method (boolean): module is now 'account_payment' ('payment')
+payment      / account.payment.method.line / payment_acquirer_id (many2one): DEL relation: payment.acquirer
+# TODO: theses fields needs to be moved to account_payment which is a new module that needs to be installed
```

Comment:
I'm not sure how to handle this. Should I:

1. move these models and fields to `account_payment` and triggering the installation of `account_payment`.
2. backup these data and make a script in `account_payment` to get these data back again.

I'm in favor of the option 1. But I'm not sure how to trigger `account_payment` installation.

Am I wrong ? How to perform that ? :)

Reply:
Hi @legalsylvain ,

So, yes, we can assume that starting from a install on 15.0, if `payment` is installed on this DB, `account` will be also. And we cannot assume that `account_payment` is installed.

I don't know when the installation of `account_payment` will be triggered during the upgrade mechanism. Also I think that `account` and `payment` can be updated in any order, as they do not depends each other.

If moving theses models and field in the `pre-migration.py` script of `payment`, what will happen if after that `account_payment` is installed (either automatically by the upgrade mechanism or manually) ? Will it drop columns and tables and create new ones, or will it understand that there is existing models and fields that already exists in the database and that does need to be created again ?

Regards

Reply:
> I don't know when the installation of account_payment will be triggered during the upgrade mechanism. Also I think that account and payment can be updated in any order, as they do not depends each other.

- 'account' depends on 'base_setup', 'product', 'analytic', 'portal', 'digest'
- 'payment' depends : 'portal'
- 'account_payment' depends on 'account', 'payment'

So, the update order will be : 
- account, ..., payment, ... account_payment
- OR payment ..., account, ... account_payment

Reply:
Ok, so I think I have enough information to decide.

The module `account_payment` will always loads after the update of `payment` module. So moving the fields in the pre-migration script of `payment` is ok. :)